nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/service/StandardControllerServiceNode.java

```diff
@@ -82,7 +84,7 @@
     private final Lock readLock = rwLock.readLock();
     private final Lock writeLock = rwLock.writeLock();
 
-    private final Set<ComponentNode> referencingComponents = new HashSet<>();
+    private final Map<String, ComponentNode> referencingComponents = new HashMap<>();
```

Comment:
What about having a set of `ComponentNode`, `PropertyDescriptor` pairs?

Reply:
Good point, I wasn't sure about that key implementation neither.

Reply:
The lookup implementation, which doesn't have a property descriptor, may be problematic